src/desktop/components/split_test/middleware.coffee

```diff
@@ -15,4 +15,13 @@ module.exports = (req, res, next) ->
       test.set v
       res.locals.sd[k.toUpperCase()] = v
 
+  # TODO: Remove when the client nav AB test ends.
+  # We hook into `process.env` and `res.locals` (sharify)
+  # to enable a seamless isomorphic experience with Reaction.
+  res.locals.sd['EXPERIMENTAL_APP_SHELL'] = Boolean(res.locals.sd['CLIENT_NAVIGATION_V3'] is 'experiment')
+  # All values set in `process.env` get stringified.
+  # So, we just want to set either _some_ truthy value here (for enabled)
+  # or not set anything at all (for disabled).
+  process.env['EXPERIMENTAL_APP_SHELL'] = true if res.locals.sd['EXPERIMENTAL_APP_SHELL']
```

Comment:
This seems to be the hacky line. We need this flag to be accessible in a pretty global context, throughout Force and Reaction. As a feature flag, this can be clearly global, feeding from ENV. Any Force server process either has this set, or doesn't have it set, for all requests.

But, in a proper AB test, this flag is essentially request-specific (whether or not that request is in the control or experiment group), which actually brings some interesting challenges. Generally, the places that were using `process.env.FLAG` can't be trivially updated to use the proper request-specific flag (`res.locals.FLAG`).

We attempted this, but won't work with concurrency (and indeed, locally I can replicate some odd and spurious behavior).

Most likely, we'll need a [slight] refactor of Reaction in order to be able to properly propagate this value everywhere it's needed.

Reply:
@mzikherman A quick fix for this would be to use [express-http-context](https://www.npmjs.com/package/express-http-context) but we'd have to merge the node 12 PR first, or upgrade to 10.4. This module doesn't work on node 10-10.3, and we're on 10.19.